$OpenBSD: patch-Modules_FindPkgConfig_cmake,v 1.4 2016/08/12 14:23:02 dcoppa Exp $

FindPkgConfig: call find_library() for every library returned by
the .pc file, so that at the end a list of full paths to the libraries
is returned.
This makes the pkg_check_modules() behaviour consistent with the
normal CMake convention, where use of link_directories() is discouraged
in favour of using absolute paths to system libraries.

Patch by Sam Thursfield <sam.thursfield@codethink.co.uk>
https://cmake.org/Bug/view.php?id=15804

--- Modules/FindPkgConfig.cmake.orig	Fri Jul 22 15:50:22 2016
+++ Modules/FindPkgConfig.cmake	Fri Aug 12 13:58:37 2016
@@ -187,6 +187,19 @@ function(_pkgconfig_add_extra_path _extra_paths_var _v
   set(${_extra_paths_var} ${${_extra_paths_var}} PARENT_SCOPE)
 endfunction()
 
+# convert library names to absolute paths
+function(_pkgconfig_find_libraries _output_var _prefix _library_names _library_dirs)
+  foreach(_libname ${_library_names})
+    # find_library stores its result in the cache, so we must pass a unique
+    # variable name for each library that we look for.
+    set(_library_path_var "${_prefix}_LIBRARY_${_libname}")
+
+    find_library(${_library_path_var} ${_libname} PATHS ${_library_dirs})
+    list(APPEND _library_paths "${${_library_path_var}}")
+  endforeach()
+  set(${_output_var} ${_library_paths} PARENT_SCOPE)
+endfunction()
+
 # scan the LDFLAGS returned by pkg-config for library directories and
 # libraries, figure out the absolute paths of that libraries in the
 # given directories, and create an imported target from them
@@ -461,6 +474,8 @@ macro(_pkg_check_modules_internal _is_required _is_sil
       _pkgconfig_invoke_dyn("${_pkg_check_modules_packages}" "${_prefix}" INCLUDE_DIRS        "(^| )-I" --cflags-only-I )
       _pkgconfig_invoke_dyn("${_pkg_check_modules_packages}" "${_prefix}" CFLAGS              ""        --cflags )
       _pkgconfig_invoke_dyn("${_pkg_check_modules_packages}" "${_prefix}" CFLAGS_OTHER        ""        --cflags-only-other )
+
+      _pkgconfig_find_libraries("${_prefix}_LIBRARIES" ${_prefix} "${${_prefix}_LIBRARIES}" "${${_prefix}_LIBRARY_DIRS}")
 
       if (_imp_target)
         _pkg_create_imp_target("${_prefix}" _no_cmake_path _no_cmake_environment_path)
